src/sentry/api/endpoints/organization_events.py

```diff
@@ -218,27 +221,30 @@ def get(self, request, organization):
             }, status=400)
 
         try:
-            tag_keys = tagstore.get_group_tag_keys_and_top_values(
+            top_values_by_key = tagstore.get_top_values_by_keys(
                 project_ids, None, environment_ids, keys=lookup_keys, get_excluded_tags=True, **snuba_args)
         except tagstore.TagKeyNotFound:
             raise ResourceDoesNotExist
 
         if non_tag_lookup_keys:
-            tag_keys.update(self.handle_non_tag_keys(non_tag_lookup_keys, snuba_args))
+            self.handle_non_tag_keys(non_tag_lookup_keys, snuba_args, top_values_by_key)
 
+        tag_keys = self._create_tag_objects(top_values_by_key)
         return Response(serialize(tag_keys, request.user))
 
-    def handle_non_tag_keys(self, keys, snuba_args):
+    def handle_non_tag_keys(self, keys, snuba_args, top_values_by_key):
         result = set([])
         for key in keys:
+            values_dict = OrderedDict()
+
             if key == 'project.name':
                 data = self._query_non_tag_data('project_id', snuba_args)
                 projects = Project.objects.filter(id__in=snuba_args['filter_keys']['project_id'])
-                for project_data in data:
-                    project = projects.filter(id=project_data['project_id'])[0]
-                    project_data['key'] = 'project.name'
-                    project_data['value'] = project.slug
-            result.add(self._create_tag_key_tag_value_objects('project', data))
+                for value in data:
+                    project_slug = projects.filter(id=value['project_id'])[0].slug
```

Comment:
Any chance of hitting a DoesNotExist error here?

Reply:
No there shouldn't be. These should have already been validated by `get_snuba_args`, which loads all the projects already to check permissions... but it only gives back the ids so I have to look them up again.